services/src/main/java/org/keycloak/services/resources/account/ApplicationResource.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.services.resources.account;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.jboss.resteasy.annotations.cache.NoCache;
+import org.jboss.resteasy.spi.HttpRequest;
+import org.keycloak.forms.account.freemarker.model.ApplicationsBean;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.services.managers.Auth;
+
+public class ApplicationResource {
+
+    private final KeycloakSession session;
+    private final UserModel user;
+
+    public ApplicationResource(KeycloakSession session, UserModel user, Auth auth, HttpRequest request) {
+        this.session = session;
+        this.user = user;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @NoCache
+    public Response getApplications() {
+        return Response.ok()
+                .entity(new ApplicationsBean(session, session.getContext().getRealm(), user).getApplications().stream().map(
+                        ApplicationRepresentation::new).collect(Collectors.toList())).build();
+    }
+
+    public static class ApplicationRepresentation extends ClientRepresentation {
+
+        private boolean internal;
+        private List<String> clientScopesGranted;
+        private List<String> additionalGrants;
+        private Map<String, List<ClientRoleRepresentation>> clientRoles;
+
+        public ApplicationRepresentation() {
+        }
+
+        public ApplicationRepresentation(ApplicationsBean.ApplicationEntry entry) {
```

Comment:
@ssilvert I just wonder if these representations would make more sense inside `org.keycloak.representations.account`. What do you think?

Reply:
@abstractj I agree.  We need to be consistent.